hw/ip/i2c/doc/i2c.md

```diff
@@ -0,0 +1,306 @@
+{{% lowrisc-doc-hdr I2C HWIP Technical Specification }}
+
+{{% regfile ../data/i2c.hjson}}
+
+{{% section1 Overview }}
+
+This document specifies I2C hardware IP functionality. This module conforms
```

Comment:
Hi @tjaychen,
   Following up: is the compromise above (<80 chars AND maximum one sentence per line) acceptable?

Reply:
Please stick with one sentence per line for consistency with all other documentation. You can configure your editor to soft-wrap at 80 characters if you prefer.

Reply:
Hi Phil, 
   Does soft-wrap mean "maximum 1 sentence per line, maximum 80 characters".  If so, that's what I /think/ I've done in the latest commit.  Could to do me a favor and take a glance at the latest commit?  I did my best to change it.  If there is still a problem, please to let me know.  I'd like to be consistent indeed.  Thanks.
-M

Reply:
No, I meant "1 sentence per line, as many characters as it takes", as mandated by the style guide. 

To make editing and viewing more convenient, most editors can soft-wrap: visually wrap at the end of the editor window or after a number of characters to avoid horizontal scrolling. That's only an editor setting, it doesn't affect the file.

Reply:
![image](https://user-images.githubusercontent.com/47870387/68877573-93daa700-0706-11ea-809d-19dc6f77389a.png)

Reply:
So does the style guide need to be changed?

Reply:
No. It's #2 for documentation like the one you're writing right now, and #1 for README files.

Reply:
To be fair, we're not consistent about that rule currently (thanks @tomroberts-lowrisc for reminding me). But since we agreed on the style guide, the path forward would be: follow it for new documentation like this one (1 sentence per line); and gradually clean up existing ones.

Reply:
Hi Phillip, 
  :)  I don't argue that it's may have mostly done differently elsewhere.  Though I think the distinction that you are referring to is not clear in the style guide at all.  There is no mention of "README" in item 1, and "DOCUMENTATION" in item 2.   Am I the only one who somehow doesn't see this distinction in the text above?
   Forgive me if I'm sounding too much like a lawyer here, but if this is not made explicit in the style guide somehow, then it doesn't seem fair to say "we agreed on the style guide".
    While we are at it could we add a motivation for this distinction?  Is there a good reason for it?  It seems very arbitrary, and potentially frustrating to anyone working on these documents.  Basically I think the style guide is very unclear on this point as written and the motivation for this is not reflected there at all.
-M

Reply:
Ah never mind I see it now.  "[This case] is recommended for README files".  Though could we just open a PR to say that case 2 is for documentation then?

I still have no idea why we make this seemingly arbitrary distinction, though.

Reply:
Please open a PR to clarify the text.

The distinction comes from the fact that some markdown files are mostly viewed in HTML rendered form (like all doc files). For those, the style guide optimized for review and editability.

Other markdown files are viewed in an editor in raw form. That's the case for README-style files. For those, the style guide optimizes for readability in unrendered form.

Reply:
I see.  Thanks.  I'll put your text here in a proposed style guide PR, include your rationale and make the guideline much more prescriptive (rather than the suggestive tone taken currently).
(Also thanks for the review.)

Reply:
Great, thanks! Once you update the style in this PR we should be all good to get this merged quickly.